### PR TITLE
refac: get grid areas df tests

### PR DESCRIPTION
**Pull request by johevemi**
## Description
Refactor test_get_grid_areas_df.py
Create assert_extension.py with a function to assert contract matches schema

## References

* #32

### CODE REVIEW

File: source/databricks/tests/assert_extension.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import json
+
+
+def assert_contract_matches_schema(contract_path, schema):
+    expected_schema = json.load(open(contract_path))["fields"]
+    actual_schema = json.loads(schema.json())["fields"]
+    # Assert: Schema and contract has the same number of fields
+    assert len(actual_schema) == len(expected_schema)
+    # Assert: Schema matches contract
```
Comment (BjarkeMeier) [inline source/databricks/tests/assert_extension.py:23]: nit: Can we add line breaks before the two assert comments?

Comment (BjarkeMeier) [inline source/databricks/tests/assert_extension.py:23]:
```suggestion

    # Assert: Schema and contract has the same number of fields
    assert len(actual_schema) == len(expected_schema)

    # Assert: Schema matches contract
```

File: source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_grid_areas_df.py [hunk 2]
```diff
@@ -84,53 +85,47 @@ def test__stored_time_matches_persister(grid_area_df_factory, source_path):
...
-        )
-    assert i["name"] == actual_field["name"]
-    assert i["type"] == actual_field["type"]
+
+    # Assert: Test data schema matches schema
+    assert (
+        raw_integration_events_df.select(col("body").cast("string"))
+        .withColumn("body", from_json(col("body"), grid_area_updated_event_schema))
+        .select(col("body.*"))
+        .schema
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_grid_areas_df.py:111]: Can this be moved to a named variable?

File: source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_grid_areas_df.py [hunk 3]
```diff
@@ -84,53 +85,47 @@ def test__stored_time_matches_persister(grid_area_df_factory, source_path):
...
+    ) == grid_area_updated_event_schema
+
+    # Assert: From previous asserts:
+    # If schema matches contract and test data matches schema and test data results in
+    # the expected row we know that the production code works correct with data that complies with the contract
+    assert _get_grid_areas_df(raw_integration_events_df, [grid_area_code]).count() == 1
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_grid_areas_df.py:117]: How about introducing an `actual` variable?

File: source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_grid_areas_df.py [hunk 4]
```diff
@@ -84,53 +80,49 @@ def test__stored_time_matches_persister(grid_area_df_factory, source_path):
...
+    )
+
+    # Assert: Test data schema matches schema
+    assert test_data_schema == grid_area_updated_event_schema
+
+    # Assert: From previous asserts:
+    # If schema matches contract and test data matches schema and test data results in
+    # the expected row we know that the production code works correct with data that complies with the contract
+    actual = _get_grid_areas_df(raw_integration_events_df, [grid_area_code]).count()
+    assert actual == 1
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_grid_areas_df.py:114]:
```suggestion
    actual_df = _get_grid_areas_df(raw_integration_events_df, [grid_area_code])
    assert actual_df.count() == 1
```

File: source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_grid_areas_df.py [hunk 5]
```diff
@@ -84,53 +80,49 @@ def test__stored_time_matches_persister(grid_area_df_factory, source_path):
...
 ):
     # Arrange
-    grid_area_updated_schema = get_from_file(
+    grid_area_updated_schema = read_contract(
         f"{source_path}/contracts/events/grid-area-updated.json"
     )
     message_type = next(
-        x for x in grid_area_updated_schema["bodyFields"] if x["name"] == "MessageType"
+        x for x in grid_area_updated_schema["fields"] if x["name"] == "MessageType"
     )["value"]
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_grid_areas_df.py:126]: Suggestion: Add `get_message_type` to contract_utils:
```
    message_type = get_message_type(grid_area_updated_schema)
```

Comment (johevemi) [reply source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_grid_areas_df.py:126]: good idea

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**